api/controllers/bookmark.js

```diff
@@ -154,32 +162,44 @@ router.post("/add", function addBookmark(req, res, next) {
  *
  */
 
-router.delete("/delete/:bookmarkID", function updateUser(req, res, next) {
+router.delete("/delete", function updateUser(req, res, next) {
   let userId = req.user.user_id;
-  let bookmarkID = parseInt(req.params.bookmarkID);
+  let bookmarktype = req.body.bookmarkType;
+  let thingid = req.body.thingID;
```

Comment:
More SQL validation and escaping needed, but I can tackle that on the branch where I'm securing SQL